worlds/metroidprime/Items.py

```diff
@@ -1,63 +1,81 @@
-    "Gravity Suit": ItemData(5031021, ItemClassification.progression),
-    "Varia Suit": ItemData(5031022, ItemClassification.progression),
-    "Phazon Suit": ItemData(5031023, ItemClassification.progression),
-    "Energy Tank": ItemData(5031024, ItemClassification.useful),
-    "Spring Ball": ItemData(5031025, ItemClassification.useful),
+suit_upgrade_table: dict[str, ItemData] = {
+    "Power Beam": ItemData("Power Beam", 0, ItemClassification.progression),
+    "Ice Beam": ItemData("Ice Beam", 1, ItemClassification.progression),
+    "Wave Beam": ItemData("Wave Beam", 2, ItemClassification.progression),
+    "Plasma Beam": ItemData("Plasma Beam", 3, ItemClassification.progression),
+    "Missile Expansion": ItemData("Missile Expansion", 4, ItemClassification.useful, 999),
+    "Scan Visor": ItemData("Scan Visor", 5, ItemClassification.progression),
+    "Morph Ball Bombs": ItemData("Morph Ball Bombs", 6, ItemClassification.progression),
+    "Power Bomb Expansion": ItemData("Power Bomb Expansion", 7, ItemClassification.useful, 99),
+    "Flamethrower": ItemData("Flamethrower", 8, ItemClassification.filler),
+    "Thermal Visor": ItemData("Thermal Visor", 9, ItemClassification.progression),
+    "Charge Beam": ItemData("Charge Beam", 10, ItemClassification.progression),
+    "Super Missile": ItemData("Super Missile", 11, ItemClassification.progression),
+    "Grapple Beam": ItemData("Grapple Beam", 12, ItemClassification.progression),
+    "X-Ray Visor": ItemData("X-Ray Visor", 13, ItemClassification.progression),
+    "Ice Spreader": ItemData("Ice Spreader", 14, ItemClassification.filler),
+    "Space Jump Boots": ItemData("Space Jump Boots", 15, ItemClassification.progression),
+    "Morph Ball": ItemData("Morph Ball", 16, ItemClassification.progression),
+    "Combat Visor": ItemData("Combat Visor", 17, ItemClassification.progression),
+    "Boost Ball": ItemData("Boost Ball", 18, ItemClassification.progression),
+    "Spider Ball": ItemData("Spider Ball", 19, ItemClassification.progression),
+    "Power Suit": ItemData("Power Suit", 20, ItemClassification.progression),
+    "Gravity Suit": ItemData("Gravity Suit", 21, ItemClassification.progression),
+    "Varia Suit": ItemData("Varia Suit", 22, ItemClassification.progression),
+    "Phazon Suit": ItemData("Phazon Suit", 23, ItemClassification.progression),
+    "Energy Tank": ItemData("Energy Tank", 24, ItemClassification.useful),
+    "Spring Ball": ItemData("Spring Ball", 25, ItemClassification.useful),
     # item 026 is a health refill
-    "Ice Trap": ItemData(5031027, ItemClassification.trap),
-    "Wavebuster": ItemData(5031028, ItemClassification.filler),
+    "Ice Trap": ItemData("Ice Trap", 27, ItemClassification.trap),
+    "Wavebuster": ItemData("Wavebuster", 28, ItemClassification.filler),
+}
+
+# These item ids are invalid in the player state, we'll need to exclude it from logic relying on that
+custom_suit_upgrade_table: dict[str, ItemData] = {
```

Comment:
AP is requiring we have these with separate ids. I was thinking I could have some dedicated cases in the item handling script that treats these 2 slightly differently since they are extensions of the base game rather than a vanilla item